test/bin/ci_phase_iso_build.sh

```diff
@@ -21,49 +20,111 @@ exec &> >(tee >(awk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush() }' >"${
 
 PULL_SECRET=${PULL_SECRET:-${HOME}/.pull-secret.json}
 
+# Allow for a dry-run option to save on testing time
+BUILD_DRY_RUN=${BUILD_DRY_RUN:-false}
+dry_run() {
+    ${BUILD_DRY_RUN} && echo "echo"
+}
+
+build_cache_update() {
+    # Build the images to be cached
+    $(dry_run) bash -x ./bin/build_images.sh -i -g ./image-blueprints/group1
+    $(dry_run) bash -x ./bin/build_images.sh -i -g ./image-blueprints/group2
+
+    # Upload the images and update the 'last' setting
+    ./bin/manage_build_cache.sh upload  -b "${SCENARIO_BUILD_BRANCH}" -t "${SCENARIO_BUILD_TAG}"
+    ./bin/manage_build_cache.sh setlast -b "${SCENARIO_BUILD_BRANCH}" -t "${SCENARIO_BUILD_TAG}"
+
+    # Cleanup older images in the cache, preserving the previous cache if any
+    # The 'last' cache is preserved by default
+    ./bin/manage_build_cache.sh keep -b "${SCENARIO_BUILD_BRANCH}" -t "${SCENARIO_BUILD_TAG_PREV}"
+}
+
+build_with_cache() {
```

Comment:
I think naming if a bit off: `_with_cache` but inside there's `if $HAS_BUILD_CACHE` so it can be "without cache" after all.

Maybe we could rename these two new functions so they hint which one we expect in presubmits and which one in special periodic?
Or do you think it would be beneficial for readability to have separate files for "testing flow" and "cache building flow"? I.e. instead of `if [ $# -gt 0 ] && [ "$1" = "-update_cache" ]`  we'd just have another script?

Reply:
I will fix the naming, no problem.
About spliting the files, I thought about it, but there will be 90% duplication of code in them, so I thought to keep it in the same script.